vsphere/datadog_checks/vsphere/data/conf.yaml.example

```diff
@@ -82,49 +60,63 @@ instances:
-    #
-    # include_only_marked: false
-
-    ## @param collection_level - integer - optional - default: 1
-    ## A number between 1 and 4 to specify how many metrics will be sent
-    ## 1: Only basic metrics - 4: every metric available.
-    ## Warning: Depending on the size of the vSphere environment, metric collection can be slow,
-    ## very CPU intensive and put pressure on the vCenter Server
-    #
-    # collection_level: 1
-
-    ## @param collect_historical_only - boolean - optional - default: false
-    ## By default the check collects metrics from realtime resources (hosts and VMs).
-    ## Set this to true to only collect metrics from historical resources
-    ## (cluster, datastore and datacenter).
-    ## In order to collect metrics from those two types of resources, create two
-    ## 'instances' in this config file pointing to the same vcenter server, one with this flag
-    ## to false, the other with this flag to true.
-    ## The reason for this requirement is that collecting historical resources decreases
-    ## the performance of the check; having two instances allows realtime metrics collection
-    ## not to be slowed down by the collection of historical metrics.
-    ## Note: An instance with this flag set to true does not submit any service check, events or host tags.
-    ##       It is designed to be used only when next to another instance of the check.
-    #
-    # collect_historical_only: false
+    ## @param resource_filters - list of objects - optional - default: no filter
+    ## Each filter in the list is composed of three parameters.
+    ## 'resource' is one of vm/host/datastore/datacenter/cluster on which to apply the filter
+    ## 'patterns' is a list of matching regex, if any of those matches, the resource will be monitored.
+    ## 'property' is the kind of property on which to apply the filter and must be one of those values:
+    ##    - 'property: name' (default) to filter on the resource name (defined by vCenter)
+    ##    - 'property: inventory_path' to filter on the full inventory path
+    ##       (ex: \/<Datacenter_name>\/vm\/<FOLDER_NAME>\/.* for all vms in the folder <FOLDER_NAME>)
+    ##    - 'property: hostname' to filter on the ESXi Host name (only valid for if 'resource' equals 'vm')
+    ##    - 'property: guest_hostname' to filter on the VM hostname (only valid for if 'resource' equals 'vm' and
+    ##       if VMware tools are installed)
+    ## Note1: 'datastore', 'datacenter' and 'cluster' filters are ignored when collecting realtime metrics as those
+    ##         resources do not have realtime metrics
+    ## Note2: 'vm' and 'host' filters are ignored when collecting historical metrics as those resources do not have
+    ##         historical metrics
+    #
+    # resource_filters:
+    #   - resource: vm
+    #     property: name
+    #     patterns:
+    #       - <VM_REGEX_1>
+    #       - <VM_REGEX_2>
```

Comment:
(nit) Should we just allow one pattern? You can usually handle all cases using regex constructs e.g. `(foo|bar)`, but it's maybe less readable...

Reply:
I think it's just easier to manage with a list instead of mixing all the patterns into one.